sdk/nanotdf_config.go

```diff
@@ -0,0 +1,66 @@
+package sdk
+
+import (
+	"crypto/ecdh"
+	"fmt"
+
+	"github.com/opentdf/platform/lib/ocrypto"
+)
+
+// ============================================================================================================
+// Support for specifying configuration information for nanoTDF operations
+//
+// The config information in this structure is referenced once at the beginning of the nanoTDF
+// operation, and is not consulted again.  It is safe to create a config, use it in one operation, modify it,
+// and use it again in a second operation.  The modification will only affect the second operation in that case.
+//
+// ============================================================================================================
+
+type NanoTDFConfig struct {
+	keyPair      ocrypto.ECKeyPair
+	kasPublicKey *ecdh.PublicKey
+	attributes   []string
+	cipher       CipherMode
+	kasURL       ResourceLocator
+	sigCfg       signatureConfig
+	policy       policyInfo
+	bindCfg      bindingConfig
+}
+
+type NanoTDFOption func(*NanoTDFConfig) error
+
+// NewNanoTDFConfig - Create a new instance of a nanoTDF config
+func (s SDK) NewNanoTDFConfig() (*NanoTDFConfig, error) {
+	// TODO FIXME - how to pass in mode value and use here before 'c' is initialized?
+	newECKeyPair, err := ocrypto.NewECKeyPair(ocrypto.ECCModeSecp256r1)
+	if err != nil {
+		return nil, fmt.Errorf("ocrypto.NewRSAKeyPair failed: %w", err)
+	}
+
+	c := &NanoTDFConfig{
+		keyPair: newECKeyPair,
+		bindCfg: bindingConfig{
+			useEcdsaBinding: false,
+			padding:         0,
+			eccMode:         ocrypto.ECCModeSecp256r1,
+		},
+		cipher: kCipher96AuthTagSize,
+		sigCfg: signatureConfig{
+			hasSignature:  false,
+			signatureMode: ocrypto.ECCModeSecp256r1,
+			cipher:        cipherModeAes256gcm96Bit,
+		},
+	}
+
+	return c, nil
+}
+
+// SetKasURL - set the URL of the KAS endpoint to be used for this nanoTDF
+func (config *NanoTDFConfig) SetKasURL(url string) error {
```

Comment:
This should follow the variadic function options pattern we setup with tdf. 

https://github.com/opentdf/platform/blob/771abd6423d15b7dc30ba4742348f0195ea36037/sdk/tdf_config.go#L100-L110

Reply:
Added functions